codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -97,9 +100,36 @@ public boolean prefaceReceived() {
 
     @Override
     public void decodeFrame(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Http2Exception {
+        // If it's the first frame we've decoded, make sure it's a SETTINGS frame.
```

Comment:
I think my previous comment still applies https://github.com/netty/netty/pull/3894#discussion_r32673264. Can we move to `PrefaceDecoder `?